packages/ipfs/test/http-api/cors.js

```diff
@@ -0,0 +1,123 @@
+/* eslint max-nested-callbacks: ["error", 8] */
+/* eslint-env mocha */
+'use strict'
+
+const { expect } = require('aegir/utils/chai')
+const http = require('../utils/http')
+const sinon = require('sinon')
+
+describe('cors', () => {
+  let ipfs
+
+  beforeEach(() => {
+    ipfs = {
+      id: sinon.stub().returns({
+        id: 'id',
+        publicKey: 'publicKey',
+        addresses: 'addresses',
+        agentVersion: 'agentVersion',
+        protocolVersion: 'protocolVersion'
+      })
+    }
+  })
+
+  describe('should allow configuring CORS', () => {
```

Comment:
This needs additional tests for [preflight `OPTIONS` request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS#Preflighted_requests_in_CORS) with `Access-Control-Request-Method: POST`.

I suspect that right now `OPTIONS` request will fail,  due to `405 Method Not Allowed`, which means CORS is broken even when all headers are in place.

Reply:
I'll add a test.  I didn't initially because Hapi doesn't allow configuring the `Access-Control-Request-Method` header that I can see.

Reply:
I've added some tests but I'm not 100% sure that go-ipfs is doing the right thing here either: https://github.com/ipfs/go-ipfs/issues/7667

Reply:
Slight aside but for future issue spelunkers Hapi doesn't allow configuring the `Access-Control-Request-Method` header, because it's sent by the client and Hapi's CORS support just [checks to see if a handler is available on the requested resource with that method](https://github.com/hapijs/hapi/blob/eb89f71df7f8ecc48559eeeffbe3414205b45691/lib/cors.js#L103), waving the request through if one exists with CORS config.